reactor-core/src/jcstress/java/reactor/core/publisher/MonoDelayStressTest.java

```diff
@@ -1,175 +1,123 @@
 		}
 
 		@Arbiter
-		public void arbiter(IIIII_Result r) {
-			r.r1 = subscriber.onNextDiscarded.get();
+		public void arbiter(IIZ_Result r) {
+			r.r1 = subscriber.onNextCalls.get();
 			r.r2 = subscriber.onErrorCalls.get();
-			r.r3 = subscriber.droppedErrors.size();
-			r.r4 = subscriptionOuter.cancelled.get() ? 1 : 0;
-			r.r5 = subscriptionInner.cancelled.get() ? 1 : 0;
 		}
 	}
 
 	@JCStressTest
-	@Outcome(id = {"1, 0, 1, 1"}, expect = ACCEPTABLE, desc = "Value discarded. Subscriptions cancelled")
-	@Outcome(id = {"0, 1, 0, 0"}, expect = ACCEPTABLE, desc = "Value delivered. Cancel signal is late")
+	@Outcome(id = {"1, 0, 1"}, expect = ACCEPTABLE, desc = "Tick was delivered, request happened before tick")
+	@Outcome(id = {"1, 0, 2"}, expect = ACCEPTABLE, desc = "Tick was delivered, request happened after tick")
+	@Outcome(id = {"0, 0, 1"}, expect = ACCEPTABLE, desc = "Tick was cancelled, request happened before tick")
+	@Outcome(id = {"0, 0, 2"}, expect = ACCEPTABLE, desc = "Tick was cancelled, request happened after tick")
+	@Outcome(id = {"0, 0, 3"}, expect = ACCEPTABLE, desc = "Tick was cancelled before any interaction")
+	@Outcome(id = {"1, 0, 3"}, expect = ACCEPTABLE_INTERESTING, desc = "Tick was cancelled before any interaction, but the delivery still happened")
 	@State
-	public static class CompleteVsCancelStressTest {
+	public static class RequestAndCancelStressTest {
 
-		final StressSubscriber<Integer> subscriber = new StressSubscriber<Integer>(1L);
+		/*
+		Implementation notes: in this test we use the VirtualTimeScheduler to better coordinate
+		the triggering of the `run` method. We also use the firstStep AtomicInteger to
+		track which actor gets to run first (1 is request, 2 is delay tick, 3 is cancel)
+		 */
 
-		StressSubscription<Integer> subscriptionOuter;
-		StressSubscription<Integer> subscriptionInner;
+		final StressSubscriber<Long> subscriber = new StressSubscriber<>(1L);
+		final VirtualTimeScheduler   virtualTimeScheduler;
+		final MonoDelay              monoDelay;
+		final AtomicInteger          firstStep = new AtomicInteger();
 
 		{
-			new Mono<Integer>() {
-				@Override
-				public void subscribe(CoreSubscriber<? super Integer> actual) {
-					subscriptionOuter = new StressSubscription<>(actual);
-					actual.onSubscribe(subscriptionOuter);
-					actual.onNext(1);
-				}
-			}
-					.delayUntil(__ -> new Mono<Integer>() {
-						@Override
-						public void subscribe(CoreSubscriber<? super Integer> actual) {
-							subscriptionInner = new StressSubscription<>(actual);
-							actual.onSubscribe(subscriptionInner);
-						}
-					})
-					.subscribe(subscriber);
+			virtualTimeScheduler = VirtualTimeScheduler.create();
+			monoDelay = new MonoDelay(1, TimeUnit.NANOSECONDS, virtualTimeScheduler);
+			monoDelay.subscribe(subscriber);
 		}
 
 		@Actor
-		public void completeOuter() {
-			subscriptionOuter.actual.onComplete();
+		public void request() {
+			firstStep.compareAndSet(0, 1);
```

Comment:
same. we may think that request happened first but this will be a false assumption since advanceTimeBy can happen earlier and what we will see in the results are incorrect